packages/fetchai/connections/p2p_libp2p/connection.py

```diff
@@ -225,14 +266,75 @@ def __init__(
         self.proc = None  # type: Optional[subprocess.Popen]
         self._log_file_desc = None  # type: Optional[IO[str]]
 
-        self._config = ""
         self.logger = logger
         self._connection_timeout = (
             connection_timeout if connection_timeout is not None else PIPE_CONN_TIMEOUT
         )
         self._max_restarts = max_restarts
         self._restart_counter: int = 0
 
+    def _make_env_file(self, pipe_in_path: str, pipe_out_path: str) -> str:
+        # setup config
+        if os.path.exists(self.env_file):
+            os.remove(self.env_file)  # pragma: nocover
+
+        config = ""
+        config += "AEA_AGENT_ADDR={}\n".format(self.address)
+        config += "AEA_P2P_ID={}\n".format(self.key)
+        config += "AEA_P2P_URI={}\n".format(str(self.uri))
+        config += "AEA_P2P_ENTRY_URIS={}\n".format(
+            ",".join(
+                [
+                    str(maddr)
+                    for maddr in self.entry_peers
+                    if str(maddr) != str(self.uri)  # TOFIX(LR) won't exclude self
+                ]
+            )
+        )
+        config += "NODE_TO_AEA={}\n".format(pipe_in_path)
+        config += "AEA_TO_NODE={}\n".format(pipe_out_path)
+        config += "AEA_P2P_URI_PUBLIC={}\n".format(
+            str(self.public_uri) if self.public_uri is not None else ""
+        )
+        config += "AEA_P2P_DELEGATE_URI={}\n".format(
+            str(self.delegate_uri) if self.delegate_uri is not None else ""
+        )
+        config += "AEA_P2P_URI_MONITORING={}\n".format(
+            str(self.monitoring_uri) if self.monitoring_uri is not None else ""
+        )
+        config += "AEA_P2P_POR_ADDRESS={}\n".format(self.record.address)
+        config += "AEA_P2P_POR_PUBKEY={}\n".format(self.record.public_key)
+        config += "AEA_P2P_POR_PEER_PUBKEY={}\n".format(
+            self.record.representative_public_key
+        )
+        config += "AEA_P2P_POR_SIGNATURE={}\n".format(self.record.signature)
+        config += "AEA_P2P_POR_SERVICE_ID={}\n".format(POR_DEFAULT_SERVICE_ID)
+        config += "AEA_P2P_POR_LEDGER_ID={}\n".format(self.record.ledger_id)
+        config += "AEA_P2P_CFG_REGISTRATION_DELAY={}\n".format(
+            str(self.peer_registration_delay)
+            if self.peer_registration_delay is not None
+            else str(0.0)
+        )
+        config += "AEA_P2P_CFG_STORAGE_PATH={}\n".format(
+            self.records_storage_path if self.records_storage_path is not None else ""
+        )
+
+        with open(self.env_file, "w") as env_file:  # overwrite if exists
+            env_file.write(config)
+
+        return config
```

Comment:
I assume this is moved from below without much change (I have only spotted the context manager to open the file at line 322 as new line)

Reply:
yes, it is